java/core/src/main/java/sleeper/core/statestore/FileReferenceStoreQueries.java

```diff
@@ -0,0 +1,137 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.core.statestore;
+
+import sleeper.core.statestore.exception.FileReferenceAssignedToJobException;
+import sleeper.core.statestore.exception.FileReferenceNotFoundException;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toMap;
+
+/**
+ * Serves queries about the data files in a Sleeper table. This includes a count of the number of references
+ * to the file, and internal references which assign all the data in the file to non-overlapping partitions.
+ */
+public interface FileReferenceStoreQueries {
+
+    /**
+     * Returns all references for files in any partition. This may return multiple references for a single file if it
+     * contains records in more than one partition.
+     * <p>
+     * This must never return references for the same file on partitions where one is the ancestor of the other. This
+     * means that every record in a file must only be referenced once.
+     *
+     * @return                     a list of all {@link FileReference}s in the Sleeper table
+     * @throws StateStoreException if query fails
+     */
+    List<FileReference> getFileReferences() throws StateStoreException;
+
+    /**
+     * Returns a stream of files that are ready for garbage collection, i.e. they have no references and the last update
+     * time is before maxUpdateTime.
+     *
+     * @param  maxUpdateTime       The latest time at which a file can have been updated in order to be garbage
+     *                             collected
+     * @return                     a stream of filenames with the matching status
+     * @throws StateStoreException if query fails
+     */
+    Stream<String> getReadyForGCFilenamesBefore(Instant maxUpdateTime) throws StateStoreException;
+
+    /**
+     * Returns all file references which are not assigned to any job.
+     *
+     * @return                     a list of {@link FileReference}s which are not assigned to any job
+     * @throws StateStoreException if query fails
+     */
+    List<FileReference> getFileReferencesWithNoJobId() throws StateStoreException;
+
+    /**
+     * Checks if files on a given partition are assigned to a certain job.
+     *
+     * @param  partitionId         the ID of the partition to query
+     * @return                     a list of {@link FileReference}s on the partition
+     * @throws StateStoreException if query fails
+     */
+    default boolean isPartitionFilesAssignedToJob(String partitionId, List<String> filenames, String jobId) throws StateStoreException {
```

Comment:
Trivial point as copied, doesn't read as good English for method name due to plural in files.

Suggestion: arePartitionFilesAssignedToJob

Change not necessary, just raised for discussion as understand why starts with is as well

Reply:
Other options would be partitionFilesAreAssignedToJob, or isEachPartitionFileAssignedToJob, or replace the parameters with an object which describes the condition and make the method name shorter. All the options seem better than how it is now, now that you mention it!